### PR TITLE
Refactor/validate utils

### PR DESCRIPTION
**Pull request by zyc09**
- [x] refactoring for #40245
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/util/_validators.py [hunk 1]
```diff
@@ -427,3 +427,70 @@ def validate_ascending(
         return validate_bool_kwarg(ascending, "ascending", **kwargs)
 
     return [validate_bool_kwarg(item, "ascending", **kwargs) for item in ascending]
+
+
+def validate_endpoints(closed):
```
Comment (jreback) [inline pandas/util/_validators.py:432]: can you type this (input & output)

File: pandas/util/_validators.py [hunk 2]
```diff
@@ -427,3 +427,70 @@ def validate_ascending(
...
+        left_closed = True
+    elif closed == "right":
+        right_closed = True
+    else:
+        raise ValueError("Closed has to be either 'left', 'right' or None")
+
+    return left_closed, right_closed
+
+
+def validate_inclusive(inclusive):
```
Comment (jreback) [inline pandas/util/_validators.py:465]: can you type

**Review by jreback — CHANGES_REQUESTED** [review]
looks really code, some comments. ping on green.

**Review by jreback — APPROVED** [review]

File: pandas/util/_validators.py [hunk 3]
```diff
@@ -427,3 +428,70 @@ def validate_ascending(
         return validate_bool_kwarg(ascending, "ascending", **kwargs)
 
     return [validate_bool_kwarg(item, "ascending", **kwargs) for item in ascending]
+
+
+def validate_endpoints(closed: Any) -> tuple[bool, bool]:
```
Comment (jreback) [inline pandas/util/_validators.py:433]: hmm this should be Optional[str]

File: pandas/util/_validators.py [hunk 4]
```diff
@@ -427,3 +428,70 @@ def validate_ascending(
...
+        left_closed = True
+    elif closed == "right":
+        right_closed = True
+    else:
+        raise ValueError("Closed has to be either 'left', 'right' or None")
+
+    return left_closed, right_closed
+
+
+def validate_inclusive(inclusive: Any) -> tuple[bool, bool]:
```
Comment (jreback) [inline pandas/util/_validators.py:466]: same here Optional[str]

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/util/_validators.py (hunk 4 above)
Comment (zyc09) [reply pandas/util/_validators.py:466]: oof, let me amend that

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @zyc09 very nice!

**Pull request merged**